### PR TITLE
update gb-frontend image. New image includes the change in PR # 23381.

### PR DESCRIPTION
**Pull request by amygdala**
Update to use the gcr.io/google-samples/gb-frontend:v4 image. New image includes the change in https://github.com/kubernetes/kubernetes/pull/23381.

### CODE REVIEW

**Comment by amygdala** [comment]
@janetkuo Janet, I assigned you rather arbitrarily -- LMK if I should bump to someone else.

File: examples/guestbook/README.md [hunk 1]
```diff
@@ -539,7 +539,7 @@ spec:
     spec:
       containers:
       - name: php-redis
-        image: gcr.io/google_samples/gb-frontend:v3
+        image: gcr.io/google-samples/gb-frontend:v4
```
Comment (janetkuo) [inline examples/guestbook/README.md]: Why is this changed from `google_samples` to `google-samples`? Do we need to change other yaml files too?

Comment (amygdala) [inline examples/guestbook/README.md]: The hyphen version used to not be properly supported by gcr, but now it is.  They both are fine now.
(the project itself is named google-samples-- but we used to have to use the underscore).

Comment (janetkuo) [inline examples/guestbook/README.md]: If so, you may want to update other yaml files in this example too. 
```
examples/guestbook/
├── all-in-one
│   ├── frontend.yaml
│   ├── guestbook-all-in-one.yaml
│   └── redis-slave.yaml
├── frontend-deployment.yaml
├── frontend-service.yaml
├── redis-master-deployment.yaml
├── redis-master-service.yaml
├── redis-slave-deployment.yaml
└── redis-slave-service.yaml
```

**Comment by amygdala** [comment]
@janetkuo Janet, I won't have cycles to revisit this PR in the near term -- could we get this one in so people can have the fix?  It not only addresses #23381, but fixes that image regression (it addresses that change back to use v3-- which loses a php-related fix).
(The two versions of the gcr path both work fine and point to the same thing).

We can file an issue to make the others consistent if you feel stronglym -- but I don't know if it's that impt as they do both work.

**Comment by janetkuo** [comment]
Sounds good. Applying the tag.

**Comment by janetkuo** [comment]
@k8s-bot unit test this

**Comment by janetkuo** [comment]
@k8s-bot unit test this issue: #24424

**Comment by janetkuo** [comment]
@k8s-bot unit test this issue #24679
